src/Randomizer.SMZ3.Tracking/Configuration/IMergeableConfig.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Randomizer.SMZ3.Tracking.Configuration.ConfigFiles;
+using Randomizer.SMZ3.Tracking.Configuration.ConfigTypes;
+
+namespace Randomizer.SMZ3.Tracking.Configuration
+{
+    public interface IMergeableConfig
+    {
+        public void Merge(IMergeableConfig other);
```

Comment:
I'd give this a different kind of spin:

- Considering most implementations share their, well, implementation, maybe this would be better suited as an abstract class, or (new in some recent C# version) default interface implementation.
  - Is it necessary to have the list and property merge functions separate? Couldn't you just do both?
- I'd add a type parameter to enforce the consistency you're expected. I think you could do `BossConfig : IMergable<BossConfig>`.
- I'd call this "MergeFrom", to signify you're adding things into the current instance (instead of into the other one, or into a new instance).

Reply:
So, I actually did try some of these things a bit.

- Can't use an abstract class because some of these already extend another class
- Combining both the list and property merges into a single function was a bit tricky, partially because detection of Lists via reflection seems a bit touchy. You can tell is a class is a subtype of List, you have to specifically check if it's a subclass of List<Type>. And even if BossInfo is of type IMergeableConfig, a List<BossInfo> cannot be cast to List<IMergeableConfig>.
- Because of the above, I needed to split functions. Also default interface implementations are kinda annoying as a class that implements an interface won't have that method accessible unless you implement it. You have to cast the object back to the interface to access it. But, that's workable.

That being said, I may be able to do some things to improve it a bit. I could perhaps introduce class attributes that dictate what type of merging needs to be done or if I add the type parameter to the interface like you suggest I may be able to use that. If the type is the same, it's a property merge. If it's different, the class is a list of that specified type and it needs to do the list merge. That way I should be able to do the cast into List<T>.

With that, I think I should be able to do the default interface implementation.

Reply:
@Vivelin Okay, I made some updates. Typing the interface definitely introduced some annoyances such as not being able to properly cast properties to the interface to be able to use the merge method, and checking if a property is a type of an generic interface seems a bit wonky as well. But, I was able to get it to work without an external class.

Still need to do the renames you requested, though.